### PR TITLE
update changelog and pin vivarium>=3.0.0

### PR DESCRIPTION
**Pull request by stevebachmeier**
### Description

- *Category*:  release
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5233

### CODE REVIEW

**Review by patricktnast — APPROVED** [review]

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,3 +1,12 @@
+**2.0.0 - 08/13/24**
+
+ - Implement Vivarium's new results processing
```
Comment (rmudambi) [inline CHANGELOG.rst:3]: I think you should use the breaking/major/minor distinctions here as well. Breaking would be results processing, major is simulation backups, and everything else is minor.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
+ - Use external layered-config-tree package instead of Vivarium ConfigTree class
+ - Raise instead of log when there are more jobs than expected from a previous run
+ - Stop installing vivarium inside the package directory during github actions
+ - Automatically run Jenkins builds on push or pull request
+ - Implement simulation backups
```
Comment (rmudambi) [inline CHANGELOG.rst:8]: I'd rephrase this as:
```
- Add ability to back up simulations periodically during runs, and to restart from these backups
```

**Review by rmudambi — APPROVED** [review]
